packages/auth/src/shared/utils.ts

```diff
@@ -7,3 +8,19 @@ export function generateNonce(): string {
 export function generateToken(): string {
   return crypto.randomBytes(10).toString('hex').toUpperCase()
 }
+
+export function generateRouteLogs(ctx: AppContext): {
+  route: typeof ctx.path
+  method: typeof ctx.method
+  params: typeof ctx.params
+  headers: typeof ctx.headers
+  requestBody: typeof ctx.request.body
+} {
+  return {
+    method: ctx.method,
+    route: ctx.path,
+    headers: ctx.headers,
+    params: ctx.params,
+    requestBody: ctx.request.body
+  }
+}
```

Comment:
I imagine there is probably some sensitive info in here somewhere.

for example from a `POST continue/{{continueId}}`:
- `requestBody.continue.access_token.value`
- `requestBody.access_token.value`

And is everything in headers OK to log or should any of it be redacted?

```json
  "headers": {
    "accept": "application/json",
    "content-type": "application/json",
    "content-digest": "sha-512=:wJHRy2GvqCYWZhLTIAdtBaDT9MRO69ldMWgdld0GO8cYVu8uop7pmilYh1xmlVDiiT8d2P1CSdyYgnbCsJIrNA==:",
    "content-length": "174",
    "signature": "sig1=:M5wb//IFQa43pZ6DfUrxcx7RCf3aQWXApk28TmhqgiUrzBbhlu0hs2/zVzXuiB7kEa1LZuq3taQT1XT5wuBDDg==:",
    "signature-input": "sig1=(\"@method\" \"@target-uri\" \"content-digest\" \"content-length\" \"content-type\");keyid=\"keyid-97a3a431-8ee1-48fc-ac85-70e2f5eba8e5\";created=1715013373",
    "user-agent": "axios/1.6.8",
    "accept-encoding": "gzip, compress, deflate, br",
    "host": "happy-life-bank-test-auth:4106",
    "connection": "keep-alive"
  },
```

If we need to [configure pino to redact](https://github.com/pinojs/pino/blob/main/docs/redaction.md) I imagine it might make sense to do it where we initialize the logger.

Reply:
This seems like a good pattern btw though. Also wondering if you think it would make sense to desstructure this where we init the pino logger so we dont have to do `...generateRouteLogs(ctx)` everywhere? Or if we dont want to always log context do a new `loggerWithCtx` or something?